src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -85,16 +85,12 @@ public App(AppConfig appConfig) {
         this.appConfig = appConfig;
 
         ExecutorService collectionThreadPool =
-                appConfig.getThreadPoolSize() >= 1
-                        ? Executors.newFixedThreadPool(appConfig.getThreadPoolSize())
-                        : Executors.newCachedThreadPool();
+                Executors.newFixedThreadPool(appConfig.getThreadPoolSize());
```

Comment:
So one of the reasons we have this is because if customers set a negative value for the thread pool size this would be the same as having an unlimited pool. I guess you want to avoid that? It is dangerous, but could be useful to debug.

Reply:
as I mentioned in the commit description: `we validate that the configured number of threads is strictly positive`. So we don't allow users to set a `0` or negative value here

Reply:
🤦‍♂️ OK then let's decide if we want to kill the feature or fix the config value.

Reply:
decision: we might implement an unbounded pool feature in the future, no need to do it now.

Reply:
discussed on slack: let's plan allowing negative values for jmxfetch `0.27.0`, only allowing strictly positive values is enough for `0.26.1` (users can set a very high value on the number of threads, and it'll be very close to using an unbounded thread pool)